tests/ha/ha_cluster_join.pm

```diff
@@ -35,12 +35,16 @@ sub run {
     diag 'Wait until cluster is initialized...';
     barrier_wait("CLUSTER_INITIALIZED_$cluster_name");
 
+    # Configure ssh key to enable ssh passwordless
+    get_root_ssh_key $node_to_join;
+
     # Try to join the HA cluster through first node
     assert_script_run "ping -c1 $node_to_join";
     type_string "ha-cluster-join -yc $node_to_join ; echo ha-cluster-join-finished-\$? > /dev/$serialdev\n";
-    assert_screen 'ha-cluster-join-password', $join_timeout;
-    type_password;
-    send_key 'ret';
+    if (check_screen('ha-cluster-join-password', 5)) {
```

Comment:
In the previous version this waited for `$join_timeout` seconds for the password prompt but with the new version it'll wait for 5 seconds. Why the change?

Reply:
Because we need the smallest value here (default is 0), so I would like to avoid scale out time here. The timeout here is just to check the password screen, not the timeout for the execution, this one is still in the `wait_serial` call.

Reply:
That's the thing. Previous version waited at most `2 * $join_timeout`, while new one will wait at most `5 + $join_timeout`. More importantly, if we don't catch the password prompt with the `check_screen()` call, test will fail as bootstrap script will not finish waiting for the password. I understand the reasoning behind the 5 seconds as that should be more than enough for the password prompt to appear, and this will make test faster, but I worry about false positives in slower VMs. Guess we could always increase `TIMEOUT_SCALE` in settings in those cases though. I'll approve.